iothub/device/src/Transport/Http/HttpTransportHandler.cs

```diff
@@ -207,14 +209,13 @@ internal async Task UploadToBlobAsync(string blobName, Stream source, Cancellati
             try
             {
                 // 2. Use SAS URI to send data to Azure Storage Blob (PUT)
-                var blob = new CloudBlockBlob(new Uri(putString));
-                Task uploadTask = blob.UploadFromStreamAsync(source, null, null, null, cancellationToken);
-                await uploadTask.ConfigureAwait(false);
+                var blockBlobClient = new BlockBlobClient(new Uri(putString));
+                var uploadTask = await blockBlobClient.UploadAsync(source, new BlobUploadOptions(), cancellationToken).ConfigureAwait(false);
 
                 notification.CorrelationId = fileUploadResponse.CorrelationId;
-                notification.IsSuccess = uploadTask.IsCompleted;
-                notification.StatusCode = uploadTask.IsCompleted ? 0 : -1;
-                notification.StatusDescription = uploadTask.IsCompleted ? null : "Failed to upload to storage.";
+                notification.IsSuccess = uploadTask.Value.LastModified.Subtract(DateTimeOffset.Now) <= new TimeSpan(100);
```

Comment:
couldn't figure out a better way to validate successful upload

Reply:
It looks like unsuccessful uploads throw:

https://github.com/Azure/azure-sdk-for-net/blob/64386052196f8f8293367c8b24dc42ae48d8a7fc/sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs#L589

Reply:
I'd recommend putting the upload operation in a try/catch block so that we always complete the SAS URI regardless of upload success

Reply:
@timtay-microsoft we already have the upload operation is try/catch and do a post operation with failed notification in case of unsuccessful upload

Reply:
You should be fine to remove this timespan check, then. If the upload call returns without throwing, you can assume it uploaded successfully.